### PR TITLE
add conclaveRepo to global gradle.properties

### PR DESCRIPTION
**Pull request by snedamle**
This PR will be merged with 1.2 release.

For this PR we aim to remove the conclaveRepo property from all the individual gradle.properties files from all the projects.

We have described in the root README that now you can add a conclaveRepo property at GRADLE_HOME/.gradle/gradle.properties file. Doing so you dont have to maintain this property at individual project level.

To override this property you can pass in the new value via command line via the -P flag.
This helps the developer set the value only in one location.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,9 +7,52 @@
...
+This sample shows you how you can use the [Tribuo Java ML](https://tribuo.org/learn/4.0/tutorials/) library to load, train many models like classification models, regression models, clustering models etc.
+
+### Initial SetUp
+
+In order to compile and run all the samples successfully, kindly download the conclave SDK
+and update the conclaveRepo property in GRADLE_HOME/.gradle/gradle.properties as shown below
+            
+    conclaveRepo=<path-to-conclave-sdk>
+
+Please note, if GRADLE_HOME is not set, it defaults to $USER_HOME/.gradle.
```
Comment (gendal) [inline README.md]: so the path would be $USER_HOME/.gradle/.gradle/gradle.properties?  (ie would .gradle appear twice?!)

File: README.md [hunk 2]
```diff
@@ -7,9 +7,52 @@
...
+### [Tribuo Tutorials](./tribuo-tutorials):
+Tribuo is a Java machine learning library, which makes it well suited to run with Conclave. This sample provides tools for classification, regression, clustering, model development, and more.
+This sample shows you how you can use the [Tribuo Java ML](https://tribuo.org/learn/4.0/tutorials/) library to load, train many models like classification models, regression models, clustering models etc.
+
+### Initial SetUp
+
+In order to compile and run all the samples successfully, kindly download the conclave SDK
+and update the conclaveRepo property in GRADLE_HOME/.gradle/gradle.properties as shown below
+            
+    conclaveRepo=<path-to-conclave-sdk>
```
Comment (gendal) [inline README.md]: specifically what should this point to?  the root of the SDK?  the repo folder itself?  perhaps give an example?

**Review by gendal — COMMENTED** [review]
could of questions. otherwise good to go!

File: README.md (hunk 2 above)
Comment (snedamle) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (snedamle) [reply README.md]: you are correct. I corrected the same.

File: README.md [hunk 3]
```diff
@@ -7,9 +7,57 @@
...
+
+In order to compile and run all the samples successfully, kindly download the conclave SDK
+and update the conclaveRepo property in GRADLE_USER_HOME/gradle.properties as shown below
+            
+    conclaveRepo=<path-to-conclave-sdk-repo>
+
+Below is a sample gradle.properties file, which shows you how to set conclaveRepo to point to the repo folder inside conclave-sdk.
+
+    cat gradle.properties
+    conclaveRepo=/Users/sneha.damle/Documents/projects/conclave/conclave-sdk-1.1/repo
```
Comment (rusheb) [inline README.md:54]: Probably shouldn't have this path here from your machine

Comment (rusheb) [reply README.md:54]: I think this is almost what we want, but there is an additional requirement to support multiple versions of Conclave on the same system.

Therefore we may have to replace `conclaveRepo` with another variable, `conclaveHome`. `conclaveHome` will be set in the global `gradle.properties`. The `conclaveHome` directory may contain multiple copies of the SDK, and the exact version for the project will be resolved using the `conclaveVersion` variable.

As @gendal suggested, I'm planning to expand on this in a document somewhere and share it around to get everyone's approval before proceeding

File: README.md [hunk 4]
```diff
@@ -7,9 +7,57 @@
...
+
+Please note, if GRADLE_USER_HOME is not set, it defaults to $USER_HOME/.gradle.
+Once set, Conclave will always pick up the conclaveRepo from gradle.properties. To override this behaviour you can pass your own custom conclave sdk path via command line as show below.
```
Comment (rusheb) [inline README.md:57]: May be helpful to link to the gradle docs here https://docs.gradle.org/current/userguide/build_environment.html

**Comment by shamsasari** [comment]
Closing this PR as the `conclaveRepo` property is no longer applicable in 1.3.

**Pull request closed without merging**